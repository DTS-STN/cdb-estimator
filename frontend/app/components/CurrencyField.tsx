import { useTranslation } from 'react-i18next';
import { NumericFormat } from 'react-number-format';

import type { InputFieldProps } from './input-field';
import { InputField } from './input-field';

import { cn } from '~/utils/tailwind-utils';

export interface CurrencyFieldProps extends Omit<InputFieldProps, 'type' | 'value' | 'defaultValue'> {
  allowNegative?: boolean;
  value?: string | number | null;
  defaultValue?: string | number | null;
}

export function CurrencyField({ allowNegative = false, maxLength = 15, ...rest }: CurrencyFieldProps) {
  const { i18n } = useTranslation(['common']);
  return (
    <NumericFormat
      customInput={InputField}
      thousandSeparator={i18n.language === 'fr' ? ' ' : ','}
      allowedDecimalSeparators={i18n.language === 'fr' ? [',', '.'] : ['.']}
      allowNegative={allowNegative}
      decimalScale={2}
      decimalSeparator={i18n.language === 'fr' ? ',' : '.'}
      fixedDecimalScale={true}
      thousandsGroupStyle={'thousand'}
      maxLength={maxLength}
      type="text"
      beforeInput={i18n.language === 'en' ? <span className="mr-1">$</span> : undefined}
      afterInput={i18n.language === 'fr' ? <span className="ml-1">$</span> : undefined}
      className={cn('inline', rest.className)}
      {...rest}
    />
  );
}
