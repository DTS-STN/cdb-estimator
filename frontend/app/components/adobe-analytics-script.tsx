import * as adobeAnalytics from '~/utils/adobe-analytics-utils';

export function AdobeAnalyticsHeadScript() {
  const { ADOBE_ANALYTICS_JQUERY_SRC, ADOBE_ANALYTICS_SRC } = globalThis.__appEnvironment;

  if (adobeAnalytics.isEnabled()) {
    return (
      <>
        <script src={ADOBE_ANALYTICS_JQUERY_SRC} />
        <script src={ADOBE_ANALYTICS_SRC} />
      </>
    );
  }
  return <></>;
}

export function AdobeAnalyticsBottomScript() {
  if (adobeAnalytics.isEnabled() && adobeAnalytics.isDebug()) {
    return <script id="gc-analytics-bottom-script">_satellite.setDebug(true);</script>;
  }
  return <></>;
}
