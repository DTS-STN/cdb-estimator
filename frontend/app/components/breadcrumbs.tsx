import type { ReactNode } from 'react';

import type { Params, Path, To } from 'react-router';

import { faChevronRight } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import type { ResourceKey, TFunction } from 'i18next';
import { useTranslation } from 'react-i18next';

import { InlineLink } from './links';

import { useLanguage } from '~/hooks/use-language';
import type { I18nRouteFile } from '~/i18n-routes';
import * as adobeAnalytics from '~/utils/adobe-analytics-utils';

export interface BreadcrumbRouteLink {
  file: I18nRouteFile;
  hash?: Path['hash'];
  lang?: Language;
  params?: Params;
  search?: Path['search'];
}

export interface BreadcrumbItem {
  labelKey?: ResourceKey;
  label?: { fr: string; en: string };
  destinationRoute?: BreadcrumbRouteLink;
  destination?: { en: To; fr: To };
}

export interface BreadcrumbsProps {
  className?: string;
  items: BreadcrumbItem[];
}

function getLabel(item: BreadcrumbItem, t: TFunction<['common'], undefined>, currentLanguage?: Language) {
  if (item.labelKey !== undefined) {
    return t(item.labelKey);
  }
  return currentLanguage === 'fr' ? item.label?.fr : item.label?.en;
}

function getDestination(item: BreadcrumbItem, currentLanguage?: Language) {
  if (item.destinationRoute !== undefined) {
    return item.destinationRoute;
  }
  return currentLanguage === 'fr' ? item.destination?.fr : item.destination?.en;
}

export function Breadcrumbs({ className, items }: BreadcrumbsProps) {
  const { t } = useTranslation(['common']);
  const { currentLanguage } = useLanguage();

  return (
    <nav id="wb-bc" className={className} property="breadcrumb" aria-labelledby="breadcrumbs">
      <h2 id="breadcrumbs" className="sr-only">
        {t('common:breadcrumbs.you-are-here')}
      </h2>
      <div className="container">
        <ol className="flex flex-wrap items-center gap-x-3 gap-y-1" typeof="BreadcrumbList">
          {items.map((item, idx) => {
            const label = getLabel(item, t, currentLanguage);
            const destination = getDestination(item, currentLanguage);
            return (
              <li key={label} property="itemListElement" typeof="ListItem" className="flex items-center">
                {idx !== 0 && <FontAwesomeIcon icon={faChevronRight} className="mr-2 size-3 text-slate-700" />}
                <Breadcrumb
                  destination={destination}
                  data-gc-analytics-customclick={adobeAnalytics.getCustomClick(`Breadcrumb:${idx} link`)}
                >
                  {label}
                </Breadcrumb>
              </li>
            );
          })}
        </ol>
      </div>
    </nav>
  );
}

function isBreadcrumbRouteLink(destination: To | BreadcrumbRouteLink): destination is BreadcrumbRouteLink {
  return typeof destination === 'object' && 'file' in destination;
}

function Breadcrumb({ children, destination, ...rest }: { children: ReactNode; destination?: To | BreadcrumbRouteLink }) {
  if (!destination) {
    return <span property="name">{children}</span>;
  }

  return isBreadcrumbRouteLink(destination) ? (
    <InlineLink
      file={destination.file}
      hash={destination.hash}
      params={destination.params}
      search={destination.search}
      property="item"
      typeof="WebPage"
      {...rest}
    >
      <span property="name">{children}</span>
    </InlineLink>
  ) : (
    <InlineLink to={destination} property="item" typeof="WebPage" {...rest}>
      <span property="name">{children}</span>
    </InlineLink>
  );
}
