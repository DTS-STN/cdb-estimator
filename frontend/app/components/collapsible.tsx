import { useId } from 'react';
import type { ComponentProps, ReactNode } from 'react';

import { cn } from '~/utils/tailwind-utils';

export type CollapsibleSummaryProps = ComponentProps<'summary'>;

export function CollapsibleSummary({ children, className, ...props }: CollapsibleSummaryProps) {
  return (
    <summary className={cn('marker:cursor-pointer hover:text-blue-800', className)} {...props}>
      <div className="ml-2 inline-block text-lg font-medium text-slate-950 underline hover:cursor-pointer">{children}</div>
    </summary>
  );
}

export interface CollapsibleProps extends ComponentProps<'details'> {
  contentClassName?: string;
  summary: ReactNode;
}

export function Collapsible({ children, contentClassName, id, summary, ...props }: CollapsibleProps) {
  const uniqueId = useId();
  const summaryId = `${id ?? uniqueId}-summary`;
  const contentId = `${id ?? uniqueId}-content`;
  return (
    <details id={id ?? uniqueId} {...props} className="my-2">
      <CollapsibleSummary id={summaryId}>{summary}</CollapsibleSummary>
      <div id={contentId} className={cn('my-2 border-l-[2px] border-slate-950 px-6 text-lg', contentClassName)}>
        {children}
      </div>
    </details>
  );
}
