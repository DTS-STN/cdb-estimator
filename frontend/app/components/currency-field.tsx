import { useTranslation } from 'react-i18next';
import { NumericFormat } from 'react-number-format';

import type { InputFieldProps } from './input-field';
import { InputField } from './input-field';

import { removeNumericFormatting } from '~/utils/string-utils';
import { cn } from '~/utils/tailwind-utils';

export interface CurrencyFieldProps extends Omit<InputFieldProps, 'type' | 'value' | 'defaultValue'> {
  allowNegative?: boolean;
  value?: string | number | null;
  defaultValue?: string | number | null;
}

export function CurrencyField({ allowNegative = false, maxLength, defaultValue, ...rest }: CurrencyFieldProps) {
  const { i18n } = useTranslation(['common']);
  const strippedDefaultValue = Number(removeNumericFormatting(defaultValue?.toString()));
  const val = !Number.isInteger(strippedDefaultValue) ? strippedDefaultValue.toFixed(2) : strippedDefaultValue;

  return (
    <NumericFormat
      customInput={InputField}
      thousandSeparator={i18n.language === 'fr' ? ' ' : ','}
      allowedDecimalSeparators={i18n.language === 'fr' ? [',', '.'] : ['.']}
      allowNegative={allowNegative}
      decimalScale={2}
      decimalSeparator={i18n.language === 'fr' ? ',' : '.'}
      fixedDecimalScale={false}
      thousandsGroupStyle={'thousand'}
      maxLength={maxLength ?? (i18n.language === 'en' ? 15 : 16)}
      type="text"
      beforeInput={i18n.language === 'en' ? <span className="mr-1">$</span> : undefined}
      afterInput={i18n.language === 'fr' ? <span className="ml-1">$</span> : undefined}
      className={cn('inline', rest.className)}
      defaultValue={val}
      {...rest}
    />
  );
}
