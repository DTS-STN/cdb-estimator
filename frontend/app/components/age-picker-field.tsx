import { Fragment, useState } from 'react';
import type { ChangeEvent, JSX, ReactNode } from 'react';

import { useTranslation } from 'react-i18next';

import { InputError } from '~/components/input-error';
import { InputHelp } from '~/components/input-help';
import { InputLabel } from '~/components/input-label';
import { InputLegend } from '~/components/input-legend';
import { useLanguage } from '~/hooks/use-language';
import { calculateAge } from '~/utils/age-utils';
import { getLocalizedMonths } from '~/utils/date-utils';
import { cn } from '~/utils/tailwind-utils';

type DatePart = 'year' | 'month';

type AriaErrorMessage = Readonly<Record<DatePart, string | undefined>>;

/**
 * Order of date parts based on language
 */
const DATE_PART_ORDER = {
  en: ['month', 'year'] as const,
  fr: ['month', 'year'] as const,
} as const satisfies Record<Language, DatePart[]>;

const inputStyles = {
  base: 'rounded-lg border-gray-500 focus:border-blue-500 focus:ring-blue-500 block focus:ring focus:outline-none',
  disabled: 'disabled:bg-gray-100 disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-70',
  error: 'border-red-500 focus:border-red-500 focus:ring-red-500',
} as const;

/**
 * Props for the AgePickerField component
 */
export interface AgePickerFieldProps {
  defaultValues?: {
    year?: number;
    month?: number;
  };
  disabled?: boolean;
  displayAge?: boolean;
  errorMessages?: {
    all?: string;
    month?: string;
    year?: string;
  };
  helpMessagePrimary?: React.ReactNode;
  helpMessagePrimaryClassName?: string;
  helpMessageSecondary?: React.ReactNode;
  helpMessageSecondaryClassName?: string;
  id: string;
  legend: ReactNode;
  names: {
    month: string;
    year: string;
  };
  required?: boolean;
}

/**
 * AgePickerField component
 *
 * @param props - Props for the AgePickerField component
 * @returns JSX.Element
 */
export const AgePickerField = ({
  defaultValues,
  disabled,
  displayAge,
  errorMessages,
  helpMessagePrimary,
  helpMessagePrimaryClassName,
  helpMessageSecondary,
  helpMessageSecondaryClassName,
  id,
  legend,
  names,
  required,
}: AgePickerFieldProps): JSX.Element => {
  const { t } = useTranslation(['common', 'estimator']);
  const { currentLanguage = 'en' } = useLanguage(); // english by default
  const currentDatePartOrder = DATE_PART_ORDER[currentLanguage];

  // Generate unique IDs for accessibility
  const baseId = `age-picker-${id}`;
  const ids = {
    wrapper: baseId,
    legend: `${baseId}-legend`,
    error: {
      all: `${baseId}-error-all`,
      month: `${baseId}-error-month`,
      year: `${baseId}-error-year`,
    },
    help: {
      primary: `${baseId}-help-primary`,
      secondary: `${baseId}-help-secondary`,
    },
  };

  // Combine IDs for aria-describedby attribute
  const ariaDescribedBy: string = [
    ids.legend,
    helpMessagePrimary ? ids.help.primary : false,
    helpMessageSecondary ? ids.help.secondary : false,
  ]
    .filter(Boolean)
    .join(' ');

  // Generate error messages for each date part
  const ariaErrorMessage: AriaErrorMessage = currentDatePartOrder.reduce((acc, datePart) => {
    const errors = [errorMessages?.all ? ids.error.all : false, errorMessages?.[datePart] ? ids.error[datePart] : false]
      .filter(Boolean)
      .join(' ');

    return { ...acc, [datePart]: errors || undefined };
  }, {} as AriaErrorMessage);

  const [month, setMonth] = useState(defaultValues?.month);
  const [year, setYear] = useState(defaultValues?.year);
  const age = month && year && calculateAge(month, year);
  const ageAdditionalInfo = (() => {
    if (age !== undefined) {
      if (age >= 65) return t('estimator:age.max-eligible-age-info');
      else if (age < 18) return t('estimator:age.min-eligible-age-info');
    }
    return undefined;
  })();

  const handleMonthChange = (event: ChangeEvent<HTMLSelectElement>) => {
    setMonth(parseInt(event.target.value));
  };

  const handleYearChange = (event: ChangeEvent<HTMLInputElement>) => {
    setYear(parseInt(event.target.value));
  };

  // Define age picker part fields
  const agePickerPartFields = {
    year: (
      <AgePickerYearField
        id={id}
        defaultValue={year}
        name={names.year}
        label={t('common:age-picker.year.label')}
        className="w-full sm:w-32"
        ariaDescribedBy={ariaDescribedBy}
        ariaErrorMessage={ariaErrorMessage.year}
        onChange={handleYearChange}
        required={required}
        disabled={disabled}
      />
    ),
    month: (
      <AgePickerMonthField
        id={id}
        defaultValue={month}
        name={names.month}
        label={t('common:age-picker.month.label')}
        placeholder={t('common:age-picker.month.placeholder')}
        className="w-full sm:w-auto"
        currentLanguage={currentLanguage}
        ariaDescribedBy={ariaDescribedBy}
        ariaErrorMessage={ariaErrorMessage.month}
        onChange={handleMonthChange}
        required={required}
        disabled={disabled}
      />
    ),
  } as const satisfies Record<DatePart, JSX.Element>;

  return (
    <div id={ids.wrapper}>
      <fieldset className="space-y-2">
        <InputLegend id={ids.legend} required={required}>
          {legend}
        </InputLegend>

        {/* Error Messages */}
        {errorMessages && Object.values(errorMessages).filter(Boolean).length > 0 && (
          <div className="space-y-2">
            {errorMessages.all && <InputError id={ids.error.all}>{errorMessages.all}</InputError>}
            {currentDatePartOrder
              .filter((datePart) => Boolean(errorMessages[datePart]))
              .map((datePart) => (
                <p key={datePart}>
                  <InputError id={ids.error[datePart]}>{errorMessages[datePart]}</InputError>
                </p>
              ))}
          </div>
        )}

        {/* Help Messages - Primary */}
        {helpMessagePrimary && (
          <InputHelp id={ids.help.primary} className={helpMessagePrimaryClassName}>
            {helpMessagePrimary}
          </InputHelp>
        )}

        {/* Date Picker Part Fields */}
        <div className="flex flex-col space-y-2 sm:flex-row sm:space-y-0 sm:space-x-2">
          {currentDatePartOrder.map((datePart) => (
            <Fragment key={datePart}>{agePickerPartFields[datePart]}</Fragment>
          ))}

          {displayAge === true && age !== undefined && year !== undefined && year > 1000 && year < 9999 && (
            <AgeDisplay age={age} additionalInfo={ageAdditionalInfo} />
          )}
        </div>

        {/* Help Messages - Secondary */}
        {helpMessageSecondary && (
          <InputHelp id={ids.help.secondary} className={helpMessageSecondaryClassName}>
            {helpMessageSecondary}
          </InputHelp>
        )}
      </fieldset>
    </div>
  );
};

/**
 * Props for the AgePickerMonthField component
 */
interface AgePickerMonthFieldProps {
  ariaDescribedBy: string;
  ariaErrorMessage?: string;
  className?: string;
  currentLanguage?: Language;
  defaultValue?: number;
  disabled?: boolean;
  id: string;
  label: string;
  name: string;
  onChange: (event: ChangeEvent<HTMLSelectElement>) => void;
  placeholder: string;
  required?: boolean;
}

/**
 * AgePickerMonthField component
 *
 * @param props - Props for the AgePickerMonthField component
 * @returns JSX.Element
 */
function AgePickerMonthField({
  ariaDescribedBy,
  ariaErrorMessage,
  className,
  currentLanguage,
  defaultValue,
  disabled,
  id,
  label,
  name,
  onChange,
  placeholder,
  required,
}: AgePickerMonthFieldProps): JSX.Element {
  const baseId = `age-picker-${id}-month`;
  const ids = {
    wrapper: baseId,
    label: `${baseId}-label`,
    select: `${baseId}-select`,
    optionUnselected: `${baseId}-option-unselected`,
    option: (monthIndex: number) => `${baseId}-option-${monthIndex}`,
  };
  const hasErrorMessage = !!ariaErrorMessage;
  const months = getLocalizedMonths(currentLanguage ?? 'en');

  return (
    <div id={ids.wrapper} className="space-y-2">
      <InputLabel id={ids.label} htmlFor={ids.select}>
        {label}
      </InputLabel>
      <select
        aria-describedby={ariaDescribedBy}
        aria-errormessage={ariaErrorMessage}
        aria-invalid={hasErrorMessage}
        aria-labelledby={ids.label}
        aria-required={required}
        className={cn(inputStyles.base, inputStyles.disabled, hasErrorMessage && inputStyles.error, className)}
        defaultValue={defaultValue}
        disabled={disabled}
        id={ids.select}
        name={name}
        onChange={onChange}
        required={required}
      >
        <option id={ids.optionUnselected} disabled hidden selected={defaultValue === undefined}>
          {placeholder}
        </option>
        {months.map((month) => {
          return (
            <option id={ids.option(month.value)} key={month.value} value={month.value}>
              {month.text}
            </option>
          );
        })}
      </select>
    </div>
  );
}

/**
 * Props for the AgePickerYearField component
 */
interface AgePickerYearFieldProps {
  ariaDescribedBy: string;
  ariaErrorMessage?: string;
  className?: string;
  defaultValue?: number;
  disabled?: boolean;
  id: string;
  label: string;
  name: string;
  required?: boolean;
  onChange: (event: ChangeEvent<HTMLInputElement>) => void;
}

/**
 * AgePickerYearField component
 *
 * @param props - Props for the AgePickerYearField component
 * @returns JSX.Element
 */
function AgePickerYearField({
  ariaDescribedBy,
  ariaErrorMessage,
  className,
  defaultValue,
  disabled,
  id,
  label,
  name,
  onChange,
  required,
}: AgePickerYearFieldProps): JSX.Element {
  const baseId = `age-picker-${id}-year`;
  const ids = {
    wrapper: baseId,
    label: `${baseId}-label`,
    input: `${baseId}-input`,
  };
  const hasErrorMessage = !!ariaErrorMessage;

  return (
    <div id={ids.wrapper} className="space-y-2">
      <InputLabel id={ids.label} htmlFor={ids.input}>
        {label}
      </InputLabel>
      <input
        aria-describedby={ariaDescribedBy}
        aria-errormessage={ariaErrorMessage}
        aria-invalid={hasErrorMessage}
        aria-labelledby={ids.label}
        aria-required={required}
        className={cn(inputStyles.base, inputStyles.disabled, hasErrorMessage && inputStyles.error, className)}
        defaultValue={defaultValue}
        disabled={disabled}
        id={ids.input}
        min={1900}
        name={name}
        required={required}
        type="number"
        inputMode="numeric"
        pattern="[0-9]*"
        maxLength={4}
        onChange={onChange}
      />
    </div>
  );
}

/**
 * Props for the AgeDisplay component
 */
interface AgeDisplayProps {
  age: number;
  additionalInfo?: string;
}

/**
 * AgeDisplay component
 *
 * @param props - Props for the AgeDisplay component
 * @returns JSX.Element
 */
function AgeDisplay({ age, additionalInfo }: AgeDisplayProps): JSX.Element {
  const { t } = useTranslation(['common']);

  return (
    <div role="alert" aria-live="assertive" className="space-y-1.5 sm:ml-30">
      <label className="block">
        <span className="font-semibold">{t('age-picker.your-age.label')}</span>
      </label>

      <div className="block max-w-prose">{`${age} ${t(age === 1 ? 'age-picker.your-age.value-suffix-singular' : 'age-picker.your-age.value-suffix-plural')}`}</div>

      <div>{additionalInfo}</div>
    </div>
  );
}
