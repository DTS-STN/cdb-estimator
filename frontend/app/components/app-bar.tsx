import { Menu } from '~/components/menu';

interface AppBarProps {
  children: React.ReactNode;
}

export function AppBar({ children }: AppBarProps) {
  return (
    <div className="bg-slate-700">
      <div className="align-center container mx-auto flex flex-wrap justify-between">
        <div className="align-center flex">
          <Menu>{children}</Menu>
        </div>
      </div>
    </div>
  );
}
