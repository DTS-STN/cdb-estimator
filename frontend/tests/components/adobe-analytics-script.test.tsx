import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import { AdobeAnalyticsBottomScript, AdobeAnalyticsHeadScript } from '~/components/adobe-analytics-script';

describe('AdobeAnalytics', () => {
  describe.skip('AdobeAnalyticsBottomScript', () => {
    it.each([
      {
        config: {
          ADOBE_ANALYTICS_ENABLED: true,
          ADOBE_ANALYTICS_DEBUG: true,
        },
      },
      {
        config: {
          ADOBE_ANALYTICS_ENABLED: true,
          ADOBE_ANALYTICS_DEBUG: false,
        },
      },
      {
        config: {
          ADOBE_ANALYTICS_ENABLED: false,
          ADOBE_ANALYTICS_DEBUG: false,
        },
      },
    ])('AdobeAnalyticsBottomScript should render the correct script tag based on config', ({ config }) => {
      globalThis.__appEnvironment = {
        ...globalThis.__appEnvironment,
        ...config,
      };
      const { container } = render(<AdobeAnalyticsBottomScript />);
      expect(container).toMatchSnapshot('expected html');
    });
  });

  describe('AdobeAnalyticsHeadScript', () => {
    it.each([
      {
        config: {
          ADOBE_ANALYTICS_ENABLED: true,
          ADOBE_ANALYTICS_SRC: 'https://assets.adobedtm.com/be5dfd287373/0127575cd23a/launch-913b1beddf7a-staging.min.js',
          ADOBE_ANALYTICS_JQUERY_SRC: 'https://code.jquery.com/jquery-3.7.1.min.js',
        },
      },
      {
        config: {
          ADOBE_ANALYTICS_ENABLED: false,
          ADOBE_ANALYTICS_SRC: 'https://assets.adobedtm.com/be5dfd287373/0127575cd23a/launch-913b1beddf7a-staging.min.js',
          ADOBE_ANALYTICS_JQUERY_SRC: 'https://code.jquery.com/jquery-3.7.1.min.js',
        },
      },
    ])('AdobeAnalyticsHeadScript should render the correct script tag based on config', ({ config }) => {
      globalThis.__appEnvironment = {
        ...globalThis.__appEnvironment,
        ...config,
      };
      const { container } = render(<AdobeAnalyticsHeadScript />);
      expect(container).toMatchSnapshot('expected html');
    });
  });
});
