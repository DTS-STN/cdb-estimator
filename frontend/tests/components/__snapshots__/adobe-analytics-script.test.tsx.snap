// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AdobeAnalytics > AdobeAnalyticsBottomScript > AdobeAnalyticsBottomScript should render the correct script tag based on config > expected html 1`] = `
<div>
  <script
    id="gc-analytics-bottom-script"
  >
    _satellite.setDebug(true);
  </script>
</div>
`;

exports[`AdobeAnalytics > AdobeAnalyticsBottomScript > AdobeAnalyticsBottomScript should render the correct script tag based on config > expected html 2`] = `<div />`;

exports[`AdobeAnalytics > AdobeAnalyticsBottomScript > AdobeAnalyticsBottomScript should render the correct script tag based on config > expected html 3`] = `<div />`;

exports[`AdobeAnalytics > AdobeAnalyticsHeadScript > AdobeAnalyticsHeadScript should render the correct script tag based on config > expected html 1`] = `
<div>
  <script
    src="https://code.jquery.com/jquery-3.7.1.min.js"
  />
  <script
    src="https://assets.adobedtm.com/be5dfd287373/0127575cd23a/launch-913b1beddf7a-staging.min.js"
  />
</div>
`;

exports[`AdobeAnalytics > AdobeAnalyticsHeadScript > AdobeAnalyticsHeadScript should render the correct script tag based on config > expected html 2`] = `<div />`;
